jsapp/js/router/router.tsx

```diff
@@ -0,0 +1,345 @@
+                />
+              }
+            />
+            <Route
+              path={ROUTES.FORM_MEDIA}
+              element={
+                <PermProtectedRoute
+                  protectedComponent={FormSubScreens}
+                  requiredPermissions={[PERMISSIONS_CODENAMES.change_asset]}
+                />
+              }
+            />
+            <Route
+              path={ROUTES.FORM_SHARING}
+              element={
+                <PermProtectedRoute
+                  protectedComponent={FormSubScreens}
+                  requiredPermissions={[PERMISSIONS_CODENAMES.manage_asset]}
+                />
+              }
+            />
+            <Route
+              path={ROUTES.FORM_RECORDS}
+              element={
+                <PermProtectedRoute
+                  protectedComponent={FormSubScreens}
+                  requiredPermissions={[PERMISSIONS_CODENAMES.manage_asset]}
+                />
+              }
+            />
+            <Route
+              path={ROUTES.FORM_REST}
+              element={
+                <PermProtectedRoute
+                  protectedComponent={FormSubScreens}
+                  requiredPermissions={[
+                    PERMISSIONS_CODENAMES.change_asset,
+                    PERMISSIONS_CODENAMES.view_submissions,
+                  ]}
+                  requireAll
+                />
+              }
+            />
+            <Route
+              path={ROUTES.FORM_REST_HOOK}
+              element={
+                <PermProtectedRoute
+                  protectedComponent={FormSubScreens}
+                  requiredPermissions={[PERMISSIONS_CODENAMES.manage_asset]}
+                />
+              }
+            />
+            <Route
+              path={ROUTES.FORM_KOBOCAT}
+              element={
+                <PermProtectedRoute
+                  protectedComponent={FormSubScreens}
+                  requiredPermissions={[PERMISSIONS_CODENAMES.manage_asset]}
+                />
+              }
+            />
+          </Route>
+
+          <Route
+            path={ROUTES.FORM_JSON}
+            element={
+              <PermProtectedRoute
+                protectedComponent={FormJson}
+                requiredPermissions={[PERMISSIONS_CODENAMES.view_asset]}
+              />
+            }
+          />
+          <Route
+            path={ROUTES.FORM_XFORM}
+            element={
+              <PermProtectedRoute
+                protectedComponent={FormXform}
+                requiredPermissions={[PERMISSIONS_CODENAMES.view_asset]}
+              />
+            }
+          />
+          <Route
+            path={ROUTES.FORM_EDIT}
+            element={
+              <PermProtectedRoute
+                protectedComponent={FormPage}
+                requiredPermissions={[PERMISSIONS_CODENAMES.view_asset]}
+              />
+            }
+          />
+        </Route>
+        <Route path='*' element={FormNotFound} />
```

Comment:
**Edit:** This wasn't quite right yet, `element=` needed to be a JSX not a function name, and then since FormNotFound is lazy-loaded, wrapped in suspense. Fixed [here](https://github.com/kobotoolbox/kpi/pull/4505/commits/968da59c86307ad2afe929c41bf7b8c101418caa).

Reply:
Yeah TS picked up a few problems. Fix seems fine.